plugin/src/main/cpp/extensions/openxr_fb_space_warp_extension_wrapper.cpp

```diff
@@ -0,0 +1,202 @@
+		space_warp_info[i].motionVectorSubImage.imageRect.extent.height = system_space_warp_properties.recommendedMotionVectorImageRectHeight;
+		space_warp_info[i].motionVectorSubImage.imageArrayIndex = i;
+
+		// @todo Update this to account for locomotion.
+		space_warp_info[i].appSpaceDeltaPose = identity_pose;
+
+		space_warp_info[i].depthSubImage.swapchain = (XrSwapchain)get_openxr_api()->openxr_swap_chain_info_get_swapchain(motion_vector_depth_swapchain_info);
+		space_warp_info[i].depthSubImage.imageRect.offset.x = 0;
+		space_warp_info[i].depthSubImage.imageRect.offset.y = 0;
+		space_warp_info[i].depthSubImage.imageRect.extent.width = system_space_warp_properties.recommendedMotionVectorImageRectWidth;
+		space_warp_info[i].depthSubImage.imageRect.extent.height = system_space_warp_properties.recommendedMotionVectorImageRectHeight;
+		space_warp_info[i].depthSubImage.imageArrayIndex = i;
+
+		space_warp_info[i].minDepth = 0.0;
+		space_warp_info[i].maxDepth = 1.0;
+
+		space_warp_info[i].farZ = get_openxr_api()->get_render_state_z_near();
+		space_warp_info[i].nearZ = get_openxr_api()->get_render_state_z_far();
+	}
+}
+
+void OpenXRFbSpaceWarpExtensionWrapper::_on_pre_render() {
+	if (!fb_space_warp_ext) {
+		return;
+	}
+
+	get_openxr_api()->openxr_swap_chain_info_acquire(motion_vector_swapchain_info);
+	get_openxr_api()->openxr_swap_chain_info_acquire(motion_vector_depth_swapchain_info);
+
+	RID motion_vector_swapchain_image = get_openxr_api()->openxr_swap_chain_info_get_image(motion_vector_swapchain_info);
+	get_openxr_api()->set_motion_vector_texture(motion_vector_swapchain_image);
+	RID motion_vector_depth_swapchain_image = get_openxr_api()->openxr_swap_chain_info_get_image(motion_vector_depth_swapchain_info);
+	get_openxr_api()->set_motion_vector_depth_texture(motion_vector_depth_swapchain_image);
+
+	int target_width = system_space_warp_properties.recommendedMotionVectorImageRectWidth;
+	int target_height = system_space_warp_properties.recommendedMotionVectorImageRectHeight;
+	Size2i render_target_size = { target_width, target_height };
+
+	get_openxr_api()->set_motion_vector_target_size(render_target_size);
+
+	Ref<OpenXRInterface> openxr_interface = XRServer::get_singleton()->find_interface("OpenXR");
+	int view_count = openxr_interface->get_view_count();
+	for (int i = 0; i < view_count; i++) {
+		space_warp_info[i].farZ = get_openxr_api()->get_render_state_z_near();
+		space_warp_info[i].nearZ = get_openxr_api()->get_render_state_z_far();
+	}
+}
+
+void OpenXRFbSpaceWarpExtensionWrapper::_on_post_draw_viewport(const RID &p_render_target) {
+	get_openxr_api()->openxr_swap_chain_info_release(motion_vector_swapchain_info);
+	get_openxr_api()->openxr_swap_chain_info_release(motion_vector_depth_swapchain_info);
+}
+
+bool OpenXRFbSpaceWarpExtensionWrapper::is_available() {
+	return available;
+}
+
+void OpenXRFbSpaceWarpExtensionWrapper::enable_space_warp(bool p_enable) {
+	if (enabled == p_enable) {
+		return;
+	}
```

Comment:
Given that we don't have any logic in this function, this `if` isn't really helpful here